### PR TITLE
Fix billing metrics tests (datapoints found error) and disable checking removed prometheus rule

### PR DESCRIPTION
**Pull request by jgarciao**
Sometimes, when we run a metrics query in OpenShift Monitoring, we get "no datapoints found" as a result. If we wait a few minutes and run the same query we get the result.

This PR adds retries to the "Run Query" keyword to avoid having this issue and also makes the Test Suite Teardown more reliable

This PR also fixes test ODS-510 after the rule _SLOs - JupyterHub DB_ was removed in build 1.1.1-41

### CODE REVIEW

**Comment by jgarciao** [comment]
Running here right now: https://opendatascience-jenkins-csb-rhods.apps.ocp4.prod.psi.redhat.com/job/rhods-ci-pr-test/163/

**Review by lugi0 — DISMISSED** [review]
lgtm

**Comment by jgarciao** [comment]
Finally it seem to work well now:
https://opendatascience-jenkins-csb-rhods.apps.ocp4.prod.psi.redhat.com/job/rhods-ci-pr-test/170/console

```
==============================================================================
Tests.Monitor And Manage.Metrics                                              
==============================================================================
Tests.Monitor And Manage.Metrics.Metrics                                      
==============================================================================
Test Existence of Prometheus Alerting Rules                           Testing Builds
Testing DeadManSnitch
Testing RHODS-PVC-Usage
Testing SLOs-haproxy_backend_http_responses_total
Testing SLOs-probe_success
| PASS |
------------------------------------------------------------------------------
Test Existence of Prometheus Recording Rules                          Testing Availability Metrics
Testing SLOs - JupyterHub
Testing SLOs - ODH Dashboard
Testing SLOs - RHODS Operator
Testing SLOs - Traefik Proxy
Testing Usage Metrics
| PASS |
------------------------------------------------------------------------------
Tests.Monitor And Manage.Metrics.Metrics                              | PASS |
2 tests, 2 passed, 0 failed
==============================================================================
Tests.Monitor And Manage.Metrics.Billing Metrics                              
==============================================================================
Verify OpenShift Monitoring results are correct when running undef... | PASS |
------------------------------------------------------------------------------
Test Billing Metric (notebook cpu usage) on OpenShift Monitoring      | PASS |
------------------------------------------------------------------------------
Tests.Monitor And Manage.Metrics.Billing Metrics                      | PASS |
2 tests, 2 passed, 0 failed
==============================================================================
Tests.Monitor And Manage.Metrics                                      | PASS |
4 tests, 4 passed, 0 failed
==============================================================================
Tests.Monitor And Manage                                              | PASS |
4 tests, 4 passed, 0 failed
```

**Review by lugi0 — DISMISSED** [review]
lgtm

File: tests/Resources/Page/OCPDashboard/Monitoring/Metrics.robot [hunk 1]
```diff
@@ -23,11 +22,20 @@ Verify Query Results Dont Contain Data
...
+  [Arguments]  ${query}  ${retry-attempts}=6
   Input Text   ${METRICS_QUERY_TEXTAREA}  ${query}
   Press Keys   ${METRICS_QUERY_TEXTAREA}    ENTER
   Wait Until Page Does Not Contain    No query entered  timeout=20
 
+  ${metrics_query_results_contain_data} =  Run Keyword and Return Status   Verify Query Results Contain Data
+  IF  not ${metrics_query_results_contain_data}
+      IF    ${retry-attempts} > 0
+        ${retry-attempts}=   Evaluate    ${retry-attempts} - 1
+        Sleep  60  reason=Waiting for results to be ready
```
Comment (vasukulkarni) [inline tests/Resources/Page/OCPDashboard/Monitoring/Metrics.robot:34]: do we need 6 mins?

Comment (jgarciao) [reply tests/Resources/Page/OCPDashboard/Monitoring/Metrics.robot:34]: I think prometheus updates the info every 5 mins

Comment (jgarciao) [reply tests/Resources/Page/OCPDashboard/Monitoring/Metrics.robot:34]: I've force pushed a new version with retry-attemps=10

**Comment by jgarciao** [comment]
New PR test with including new cluster creation:
https://opendatascience-jenkins-csb-rhods.apps.ocp4.prod.psi.redhat.com/job/rhods-ci-pr-test/173/

**Comment by jgarciao** [comment]
Testing again in new cluster: https://opendatascience-jenkins-csb-rhods.apps.ocp4.prod.psi.redhat.com/job/rhods-ci-pr-test/176

**Review by vasukulkarni — APPROVED** [review]

**Pull request merged**